src/subscription/__tests__/subscribe-test.js

```diff
@@ -933,4 +986,59 @@ describe('Subscription Publish Phase', () => {
       value: undefined,
     });
   });
+
+  it('should resolve GraphQL error from source event stream', async () => {
+    const erroringEmailSchema = emailSchemaWithResolvers(
+      async function*() {
+        yield { email: { subject: 'Hello' } };
+        throw new GraphQLError('test error');
```

Comment:
this is sort of silly, though.

i wonder if we should just drop `reportGraphQLError` there entirely; the case where the async iterable throws a graphql error seems exceedingly unlikely